ts/create-kpt-functions/src/cmd/type_create.ts

```diff
@@ -87,6 +67,73 @@ export async function typeCreate(packageDir: string) {
   log(format.finishMarker(desc));
 }
 
+// chooseTypeSource lets the user choose the source for the kubernetes types and returns
+// numeric choice
+function chooseTypeSource(): number {
+  const defaultContext = 0;
+  const ch = validator.getValidString(
+    () =>
+      question(
+        `> Kubernetes Types used to generate Typescript library (0) \n` +
+          `[0] Use built-in types (No cluster required)?\n` +
+          `[1] Select a context from kubeconfig?\n` +
+          `Please enter your numeric choice: `
+      ),
+    validator.isEmptyOrMaxInt(1),
+    defaultContext.toString()
+  );
+
+  return Number(ch);
+}
+
+// getOptsForTypeSource gets the url request options based on the numeric choice input
+// refer to chooseTypeSource() for choice outcomes
+function getOptsForTypeSource(typeSource: number): request.Options {
+  let url = BUILTIN_OPENAPI_URL;
+  let opts: request.Options = {
+    url,
+  };
+
+  switch (typeSource) {
+    case 0:
+      log('Using built-in kubernetes types');
+      // use default BUILTIN_OPENAPI_URL
```

Comment:
return url here, and throw an exception for the default case. This is a defensive programming in case you add options in the future, its fails in an obvious way.

Reply:
Done.